### DISCUSSION TITLE
How to apply retention labels

### DISCUSSION THREAD

**Discussion by sabrinadeveloper2023** [pnp/pnpcore#1160] [Q&A]
Hi,

I need to implement a function to apply retention labels in a file/record in SharePoint.

Is there a way to do this using PnP Core?

Retention labels are part of the Record Management component in SharePoint.

Thanks!

**Comment by jansenbe** [comment]
You can do this via the `SetComplianceTag` methods, see https://pnp.github.io/pnpcore/api/PnP.Core.Model.SharePoint.IListItem.html#PnP_Core_Model_SharePoint_IListItem_SetComplianceTagAsync_System_String_System_Boolean_System_Boolean_System_Boolean_System_Boolean_ for information on the method. See https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core.Test/SharePoint/ListItemTests.cs#L5047-L5095 for a sample on how to use this
